lib/new_relic/agent/transaction/distributed_tracer.rb

```diff
@@ -75,6 +75,7 @@ def log_request_headers(headers, direction = "OUTGOING")
 
         def insert_headers(headers)
           return unless NewRelic::Agent.agent.connected?
+          return unless dt_enabled?
```

Comment:
This line was not present previously, so it could change the behavior of the other operations that call #insert_headers. I'm a bit shocked that we would still insert headers on something if distributed tracing isn't enabled, so I would advocate for keeping this new LOC, however this is open to discussion!

Reply:
I think that's a great point. I've removed the line and will put this on our list for a future innovation meeting.